src/opm/io/eclipse/ERst.cpp

```diff
@@ -27,27 +27,33 @@
 #include <stdexcept>
 #include <string>
 
-#include <boost/regex.hpp>
+#include <iostream>
+
 
 namespace {
     int seqnumFromSeparateFilename(const std::string& filename)
     {
-        const auto re = boost::regex {
-            R"~(\.[FX]([0-9]{4})$)~"
-        };
-
-        auto match = boost::smatch{};
-        if (boost::regex_search(filename, match, re)) {
-            return std::stoi(match[1]);
+        int p=0;
+
+        std::string errMessage="Unable to Determine Report Step Sequence Number From Restart Filename \"" + filename + '"'; 
+        
+        if (filename.find(".F") != std::string::npos) {
+            p = filename.find_last_of(".F");
+        } else if (filename.find(".X") != std::string::npos) {
+            p = filename.find_last_of(".X");
+        } else {
+            OPM_THROW(std::invalid_argument, errMessage);
+        }
+        
+        try {
+            return std::stoi(filename.substr(p+1,filename.size() - p-1));
```

Comment:
You don't need the second argument here.  `filename.substr(p + 1)` will extract the same substring.